wdl-format/src/metadata.rs

```diff
@@ -0,0 +1,333 @@
+//! A module for formatting metadata sections (meta and parameter_meta).
+
+use wdl_ast::v1::LiteralNull;
+use wdl_ast::v1::MetadataArray;
+use wdl_ast::v1::MetadataObject;
+use wdl_ast::v1::MetadataObjectItem;
+use wdl_ast::v1::MetadataSection;
+use wdl_ast::v1::MetadataValue;
+use wdl_ast::v1::ParameterMetadataSection;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+
+use super::comments::format_inline_comment;
+use super::comments::format_preceding_comments;
+use super::first_child_of_kind;
+use super::formatter::SPACE;
+use super::Formattable;
+use super::Formatter;
+use super::NEWLINE;
+
+impl Formattable for LiteralNull {
+    fn format<T: std::fmt::Write>(
+        &self,
+        writer: &mut T,
+        _state: &mut Formatter,
+    ) -> std::fmt::Result {
+        write!(writer, "{}", self.syntax())
+    }
+}
+
+impl Formattable for MetadataObject {
+    fn format<T: std::fmt::Write>(
+        &self,
+        writer: &mut T,
+        formatter: &mut Formatter,
+    ) -> std::fmt::Result {
+        format_preceding_comments(
+            &SyntaxElement::from(self.syntax().clone()),
+            writer,
+            formatter,
+            false,
+        )?;
+
+        let open_brace = first_child_of_kind(self.syntax(), SyntaxKind::OpenBrace);
+        format_preceding_comments(&open_brace, writer, formatter, true)?;
```

Comment:
At this point, I've seen this pattern of formatting the preceding comment,
formatting the element, and then formatting the inline comment enough that I'm
wondering if this should just be abstracted out to its own method. Whatever is
supposed to be sandwiched between the two could be provided in a closure.

Reply:
Alright I partially implemented this (well, fully implemented but only applied to `metadata.rs`): https://github.com/stjude-rust-labs/wdl/pull/133/commits/5660c1e3e487591241e000419d60ff52f425bc31

Did I do this as you had envisioned? Want to check before carrying it through to the rest of the PR, as I'm not sure this is what you were asking for.

One thing to note is that it can only be used where `format_preceding_comments` and `format_inline_comments` are called with the same element. This pattern you've noticed does pop up frequently, but sometimes the elements in each of those calls is different (because we have to go up or down a level in the CST), in which case I don't think there's a good abstraction.